status-EIPs.md

```diff
@@ -220,6 +221,14 @@ Used for: Security (dont reuse wallet key) and user experience (dont request key
 Related: https://github.com/status-im/status-react/issues/9088 https://github.com/status-im/status-react/pull/9096  
 Sourcecode: https://github.com/status-im/status-react/blob/develop/src/status_im/constants.cljs#L242  
 
+### EIP1459 - Node Discovery via DNS
+
+Support: -
+Reference: https://eips.ethereum.org/EIPS/eip-1459
+Description: Allows the storing and retreving of nodes through merkle trees stored in TXT records of a domain.
+Used for: Finding Waku nodes.
```

Comment:
should this be `Waku` nodes?

Reply:
EIP-1459 is used to find bootstrap nodes, do we definitely want to use it to hardcode Waku nodes too? Does it make sense to separate these?

also cc @adambabik

Reply:
If we want to have a fully-fledged peers discovery mechanism then we would put there only bootnodes which would provided waku nodes.

The options we have:
1. Hardcoded static list of Waku nodes -- changing it requires a new app release,
2. A list of nodes taken from https://fleets.status.im/ -- periodically update the cached list of nodes,
3. Waku nodes in DNS -- like (2) but DNS based instead of HTTPS,
4. Bootnodes in DNS and a peers discovery mechanism.

In reasonable amount of time, only 1-3 are possible. (2) seems like the easiest.

Reply:
I prefer `3`, so that this is the bootstrap mechanism fully. @oskarth what are your opinions? I wrote it similar to option `2` in the spec right now however.

Reply:
@cammellos opinion appreciated here too.

Reply:
3 is also my preferred option, the only thing is that I would have 3 ideally as an additional (and preferred) mechanism to 1.

If 3 fails for whatever reason (DNS is filtered, DNS entry is corrupted/not available etc), we fallback on a static IP address.

That also helps with the fact that if a user wants to set up they own bootnodes, they might not add a DNS entry and use plain IP addresses for that, so having both mechanism is desirable.

Essentially: 
definitely 3 , ideally 3 with fallback/in addition to 1

Reply:
Gracefully I'd do it as follows:

a) 3 now with 1 as fall back / alternative 
b) Ensure there's a clean path to 4, since that's a more permissionless and proper setup with Discovery v5 (or similar). In that world, DNS would only host bootstrap nodes, and Waku nodes would be discovered (and likely persisted once found)

Reply:
Ok, to fully understand: the preferred option is to put Waku nodes in DNS. In the future, we will seamlessly change them to bootnodes and status-go implementation accordingly. We also want to have a mechanism to load Waku nodes from a static file which will be periodically updated in status-go from https://fleets.status.im/. There should be a way to pass a file as a text or as an HTTPS URL, both provided as bindings to status-react.

Reply:
@adambabik I don't see the use of adding waku nodes into the static file anymore if we use DNS discovery.